docs/api_compatibility.md

```diff
@@ -0,0 +1,47 @@
+# API compatibility
+
+JAX is constantly evolving, and we are not yet ready to commit to a stable API.
```

Comment:
This statement looks scarier than the reality. In reality, we remove APIs very rarely, and we think carefully before reducing functionality or making semantic changes. One can get a sense of the frequency of such changes by scanning the CHANGELOG. I think that this is worth pointing out.

I would also rewrite "not yet ready to commit to a stable API" to something like "and we want to be able to make improvements to the APIs".